{
  ParticipantCompletionCoordinatorRules.setParticipantCount(3);
  try {
    uba.begin();
    client1.saveData(DO_COMPLETE);
    client2.saveData(APPLICATION_EXCEPTION);
    Assert.fail("Exception should have been thrown by now");
  }
 catch (  TestApplicationException e) {
  }
  try {
    client3.saveData(DO_COMPLETE);
    uba.close();
    Assert.fail("Exception should have been thrown by now");
  }
 catch (  com.arjuna.wst.TransactionRolledBackException tre) {
  }
  assertEventLogClient1(CONFIRM_COMPLETED,COMPENSATE);
  assertEventLogClient2(CANCEL);
  assertEventLogClient3(CONFIRM_COMPLETED,COMPENSATE);
}
